File: webserver/src/main/java/com/nedap/workshop/java/model/Employee.java
package com.nedap.workshop.java.model;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Version;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

/**
 * @author paul.vandenberg
 */
@Entity
public class Employee implements Serializable {

    public Employee() {}

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(updatable = false, nullable = false)
    private Long id;

    @Version
    private int version;

    @NotNull
    @Size(min = 1, max=10)
    private String name;

    public Long getId() {
        return id;
    }

    public int getVersion() {
        return version;
    }

    public Employee(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Employee{" +
                "id=" + id +
                ", version=" + version +
                ", name='" + name + '\'' +
                '}';
    }
}

File: webserver/src/main/scala/com/nedap/workshop/scala/model/Employee.scala
package com.nedap.workshop.scala.model

import java.io.Serializable
import javax.persistence._
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

import scala.beans.BeanProperty

/**
  * @author paul.vandenberg
  */
@Entity class Employee(_name: String) extends Serializable {

  def this() = this(null)

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @Column(updatable = false, nullable = false)
  @BeanProperty
  var id: Long = _

  @Version
  @BeanProperty
  var version: Int = _

  @NotNull
  @Size(min = 1, max = 10)
  @BeanProperty
  var name: String = _name

  override def toString: String = "Employee{" + "id=" + id + ", version=" + version + ", name='" + name + '\'' + '}'
}
